src/extension/VSProject.cs

```diff
@@ -187,17 +187,23 @@ private void Load()
                 ThrowInvalidFileType( ProjectPath );
 
             StreamReader rdr = new StreamReader(ProjectPath, System.Text.Encoding.UTF8);
-            
+
             try
             {
                 _doc = new XmlDocument();
-                _doc.Load( rdr );
+                _doc.Load(rdr);
 
-                string extension = Path.GetExtension( ProjectPath );
+                string extension = Path.GetExtension(ProjectPath);
 
-                switch ( extension )
+                switch (extension)
                 {
                     case ".csproj":
+                        // We try legacy project first, then new format for .NET Core projects
+                        if (!TryLoadLegacyProject())
+                            if (!TryLoadDotNetCoreProject())
```

Comment:
LoadMSBuildProject() throws exception if you try loading a .NET Core project file. That is what trigger this PR. I didn't wan to touch LoadMSBuildProject() so the only option left was trying the new project file format before the old format.